packages/synapse-react-client/src/utils/jsonschema/getRequiredAttributes.ts

```diff
@@ -0,0 +1,42 @@
+import isArray from 'lodash-es/isArray'
+import isObject from 'lodash-es/isObject'
+import { JSONSchema7 } from 'json-schema'
+
+/**
+ * Returns a de-duplicated list of required attribute names from a JSON Schema.
+ * Traverses nested combinators (allOf, anyOf, oneOf).
+ * For:
+ *  - allOf: union of required fields
+ *  - anyOf / oneOf: union (pragmatic; intersection could be too restrictive for most UI uses)
```

Comment:
The union of required properties specified in `anyOf`/`oneOf` is misleading for the end user (they are not all actually required), so we shouldn't report it.